dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/cloudera/manager/AbstractClouderaYarnApplicationTask.java

```diff
@@ -0,0 +1,131 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.edwmigration.dumper.application.dumper.connector.cloudera.manager;
+
+import com.google.common.base.Preconditions;
+import com.google.edwmigration.dumper.application.dumper.connector.cloudera.manager.dto.ApiYARNApplicationDTO;
+import com.google.edwmigration.dumper.application.dumper.connector.cloudera.manager.dto.ApiYARNApplicationListDTO;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.function.Consumer;
+import javax.annotation.Nullable;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.impl.client.CloseableHttpClient;
+
+public abstract class AbstractClouderaYarnApplicationTask extends AbstractClouderaManagerTask {
+  private static final DateTimeFormatter isoDateTimeFormatter =
+      DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
+
+  private final String fromDate;
+
+  public AbstractClouderaYarnApplicationTask(String fileName, int includedLastDays) {
+    super(String.format("%s-%dd.jsonl", fileName, includedLastDays));
+    Preconditions.checkArgument(
+        includedLastDays > 1,
+        String.format("Amount of days must be a positive number. Get %d.", includedLastDays));
+    this.fromDate = buildISODateTime(includedLastDays);
+  }
+
+  private String buildISODateTime(int deltaInDays) {
+    ZonedDateTime dateTime =
+        ZonedDateTime.of(LocalDateTime.now().minusDays(deltaInDays), ZoneId.of("UTC"));
+    return dateTime.format(isoDateTimeFormatter);
+  }
+
+  class PaginatedClouderaYarnApplicationsLoader {
+    private final String host;
+    private final CloseableHttpClient httpClient;
+    private final int limit;
+    private int offset;
+
+    public PaginatedClouderaYarnApplicationsLoader(ClouderaManagerHandle handle) {
+      this.host = handle.getApiURI().toString();
+      this.httpClient = handle.getHttpClient();
+      this.limit = 500;
+      this.offset = 0;
+    }
+
+    public int load(String clusterName, Consumer<List<ApiYARNApplicationDTO>> callback) {
+      return load(clusterName, null, callback);
+    }
+
+    public int load(
+        String clusterName,
+        @Nullable String appType,
+        Consumer<List<ApiYARNApplicationDTO>> callback) {
```

Comment:
In general callback is fine, but you already introduced `abstract` class, so now you can introduce abstract method `writeBatch` or `writePage` and remove callback from method parameters.

Reply:
uh, the load is in separate class, not int `*ClouderaYarnApplicationTask`. Let's keep callback in this case.

Reply:
`callback` is too generic, let's call it `onPageSuccessLoad` or something like that.